File: packages/react-native-nitro-test/ios/HybridChild.swift
//
//  HybridChild.swift
//  react-native-nitro-test
//
//  Created by Marc Rousavy on 03.10.24.
//

class HybridChild: HybridChildSpec {
  var baseValue: Double {
    return 20
  }
  var childValue: Double {
    return 30
  }

  func bounceVariant(variant: NamedVariant) throws -> NamedVariant {
    return variant
  }

  func toString() -> String {
    return "HybridChild custom toString() :)"
  }
}

File: packages/react-native-nitro-test/android/src/main/java/com/margelo/nitro/test/HybridChild.kt
package com.margelo.nitro.test

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridChild : HybridChildSpec() {
  override val baseValue: Double
    get() = 20.0
  override val childValue: Double
    get() = 30.0

  override fun bounceVariant(variant: NamedVariant): NamedVariant {
    return variant
  }

  override fun toString(): String {
    return "HybridChild custom toString() :)"
  }
}
